apps/src/templates/census2017/CensusMapReplacement.jsx

```diff
@@ -0,0 +1,408 @@
+      case 'HISTORICAL_NO':
+        censusMessage =
+          'We believe this school historically offered limited or no Computer Science opportunities.';
+        color = 'blue';
+        break;
+      case 'MAYBE':
+      case 'HISTORICAL_MAYBE':
+        censusMessage = 'We have conflicting data for this school.';
+        color = 'yellow';
+        break;
+      default:
+        censusMessage = 'We need data for this school.';
+        missingCensusData = true;
+        color = 'white';
+    }
+
+    const schoolDropdownOption = {
+      value: this.props.schoolId,
+      label: `${this.props.schoolName} - ${this.props.city}, ${
+        this.props.state
+      }`,
+      school: {
+        nces_id: this.props.schoolId,
+        name: this.props.schoolName,
+        city: this.props.city,
+        state: this.props.state,
+        latitude: this.props.location.split(',')[0],
+        longitude: this.props.location.split(',')[1]
+      }
+    };
+
+    const colorClass = `color-small ${color}`;
+
+    return (
+      <div id="census-info-window" className="census-info-window">
+        <h4>
+          <b>{this.props.schoolName}</b>
+          <br />({this.props.city}, {this.props.state})
+        </h4>
+        <hr />
+        <div className="census-message">
+          <div className={colorClass} />
+          {censusMessage}
+          {!missingCensusData && (
+            <span>
+              &nbsp;
+              <a href="/yourschool/about" target="_blank">
+                (Why?)
+              </a>
+            </span>
+          )}
+        </div>
+        <div className="button-container">
+          <div className="button-link-div">
+            <a
+              onClick={() =>
+                this.props.onTakeSurveyClick(schoolDropdownOption, false)
+              }
+            >
+              <div className="button">
+                <div className="button-text">
+                  Take the survey for this school
+                </div>
+              </div>
+            </a>
+          </div>
+          <div className="button-link-div">
+            <a href="/yourschool/letter" target="_blank">
+              <div className="button">
+                <div className="button-text">
+                  Send the survey to a teacher at this school
+                </div>
+              </div>
+            </a>
+          </div>
+        </div>
+        {!missingCensusData && (
+          <div className="inaccuracy-link">
+            <a
+              onClick={() =>
+                this.props.onTakeSurveyClick(schoolDropdownOption, true)
+              }
+            >
+              I believe that the categorization for this school is inaccurate.
+            </a>
+          </div>
+        )}
+      </div>
+    );
+  }
+}
+
+export default class CensusMapReplacement extends Component {
+  static propTypes = {
+    onTakeSurveyClick: PropTypes.func.isRequired,
+    school: PropTypes.object
+  };
+
+  map = undefined;
```

Comment:
Should these all have `var`?

Reply:
Honestly not sure but they don't in the previous implementation (https://github.com/code-dot-org/code-dot-org/blob/15823fe1bf61b0f071c89995cefe0ee23edfed04/apps/src/templates/census2017/CensusMap.jsx#L137)